CONTRIBUTING.md

```diff
@@ -69,3 +69,87 @@ header.
 6. Create a pull request against the branch `master`.
 7. Be prepared to get some comments about your code and to modify it. Tip: Use
    `git rebase -i origin/master` to modify chains of commits.
+
+## Adding a test
+
+The test suite is a little ad-hoc, so the way tests work is different for each program being tested.
+
+Feel free to modify how the test scripts work, 
+
+### RGBASM
+
+Each `.asm` file corresponds to one test.
+RGBASM will be invoked on the `.asm` file.
+
+If a `.out` file exists, RGBASM's output (`println` etc.) must match its contents.
+If a `.err` file exists, RGBASM's errors (`warn`, errors, etc.) must match its contents.
+
+If a `.out.bin` file exists, the object file will be linked, and the generated ROM truncated to the length of the `.out.bin` file.
+After that, the ROM must match the `.out.bin` file.
+
```

Comment:
Please also explain that you can generate your output files something like this:
```bash
# Enter the test/asm directory
pushd tests/asm
# Generate `.out` and `.err` files
../../rgbasm -Weverything -o tmp.o mytest.asm >mytest.out 2>mytest.err
# Generate `.out.bin` file
../../rgblink -x -o mytest.out.bin tmp.o
# Delete mytest.out and/or mytest.err if they have no content
! grep -q . mytest.out && rm mytest.out
! grep -q . mytest.err && rm mytest.err
# Cleanup
rm tmp.o
popd
```

Reply:
I'd rather direct people at creating empty files, checking if the output makes sense, and copy-pasting it if so. I think it's better, because it provides some time to pause and think about that.